documents/api/viewsets.py

```diff
@@ -37,22 +37,81 @@
     CreateAttachmentSerializer,
     DocumentSerializer,
 )
-from ..serializers.document import DocumentMetadataSerializer
+from ..serializers.document import DocumentMetadataSerializer, GDPRSerializer
 from ..serializers.status_history import StatusHistorySerializer
 from ..utils import get_decrypted_file
 from .docs import (
     attachment_viewset_docs,
+    document_gdpr_viewset,
     document_metadata_viewset_docs,
     document_viewset_docs,
 )
 from .filtersets import DocumentFilterSet, DocumentMetadataFilterSet
 from .querysets import (
     get_attachment_queryset,
+    get_document_gdpr_data_queryset,
     get_document_metadata_queryset,
     get_document_queryset,
 )
 
 
+@extend_schema_view(**document_gdpr_viewset)
+class GDPRDataViewSet(AuditLoggingModelViewSet):
+    queryset = Document.objects.none()
+    serializer_class = GDPRSerializer
+    pagination_class = None
+    filter_backends = [DjangoFilterBackend]
+    lookup_field = "user__uuid"
+    http_method_names = ["get", "delete"]
+
+    def get_queryset(self):
+        user = self.request.user
+        if user.is_anonymous:
```

Comment:
I suppose there are permission classes for this

Reply:
This follows the code style of the project.